src/hooks/useKeyboardShortcut/useKeyboardShortcut.ts

```diff
@@ -0,0 +1,153 @@
+// This hook was stolen from https://github.com/arthurtyukayev/use-keyboard-shortcut
```

Comment:
Can you/do you need to include the license in full here?

__________

MIT License

Copyright (c) 2021 Arthur Tyukayev

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.

Reply:
is there no npm package for this? I'd be a bit wary of copy-pasting this much, with advanced functionality, without any tests—I think it creates a serious liability for us.

Reply:
there is an npm package, but they're using a weird ts setup that conflicts with ours. so I just copy pasted it. The hook didn't seem that complicated 🤷

Reply:
Also, we could do tests for it. 🤷

Reply:
Maybe there is a test you could copy over? To you—a React specialist, it's not complicated, but for someone who's less experienced with React/DOM it can pose a challenge. And not having tests means it's fragile. Basically—short term this is neat, long term this is burden. Ideally, we'd either add test coverage, or rewrite this hook to have only the functionality we need, and we add our tests. Otherwise, this is immediate tech debt IMHO.

Reply:
got rid of this, re-configured jest and made a module work. it's cool now 👍